Pod/Classes/SEGAnalytics.m

```diff
@@ -158,8 +158,19 @@ - (void)handleAppStateNotification:(NSNotification *)note
                             };
     });
     SEL selector = NSSelectorFromString(selectorMapping[note.name]);
-    if (selector)
-        [self callIntegrationsWithSelector:selector arguments:nil options:nil sync:true];
+  if (selector) {
+    if ([note.name isEqualToString:UIApplicationDidFinishLaunchingNotification]) {
+      while (!self.initialized) {
+        [[NSRunLoop currentRunLoop] runMode:NSDefaultRunLoopMode beforeDate:[NSDate dateWithTimeIntervalSinceNow: 1.0]];
+      }
+    }
+    NSArray *arguments = nil;
+    if (note != nil) {
+      arguments = @[note];
```

Comment:
Should we update the definitions here to indicate that a dictionary might be passed in as an argument? https://github.com/segmentio/analytics-ios/blob/master/Pod/Classes/Integrations/SEGIntegration.h#L62-L67